Support running SRO on vanilla k8s

dagrayvid opened the pull request:
This PR changes SRO to support running on vanilla k8s. For the openshift specific resources, we can use a new function `clients.HasResource` to check if it is available, and warn (but not fail) if it is not found, for instance,
```
clusterOperatorAvailable, err := clients.HasResource(configv1.SchemeGroupVersion.WithResource("clusteroperators"))
...
if clusterOperatorAvailable {
	// ... update status
} else {
	log.Info("Warning: ClusterOperator resource not available. Can be ignored on vanilla k8s.")
}

```

For the cases where the needed resource is hard to specify for example in SetUpWithManager in  controllers/specialresource_controller.go , we default to checking for the availability of the BuildConfigs API to determine whether or not we should use OCP specific resources.

This PR also adds a Make target to allow a user to configure the Deployment / kustomize .yaml for deploying on Vanilla, which just involves disabling the clusterOperator, prometheus ServiceMonitor and the kube-rbac-proxy used for the /metrics endpoint. In the case where a user wants to use these APIs and has them installed on a vanilla k8s cluster they can be enabled manually by commenting/uncommenting  lines in `config/default/kustomization.yaml`.

dagrayvid committed:
Enabling vanilla-k8s with checks for vanilla/OCP and centos-simple-kmod CR

dagrayvid committed:
Add auth-proxy-patch to Prometheus check in Makefile

dagrayvid committed:
Working on OCP and SRO, with bad check for OnOCP()

dagrayvid committed:
Rewrite of checks for on OCP

dagrayvid committed:
Rewrite of OCP/Vanilla k8s checks to depend on specific resources instead of NFD labels

dagrayvid committed:
Clean up Makefile for vanilla-k8s

dagrayvid committed:
Rebase

dagrayvid committed:
Update charts after rebase

zvonkok commented on controllers/resources.go:
```diff
@@ -50,6 +51,14 @@ func getChartTemplates(r *SpecialResourceReconciler) (*unstructured.Unstructured
 
 func createImagePullerRoleBinding(r *SpecialResourceReconciler) error {
 
+	_, err := clients.Interface.RbacV1().ClusterRoles().Get(context.TODO(), "system:image-puller", metav1.GetOptions{})
```
clients.Interface.Get did not work?

zvonkok replied:
Why is the handling not done in line 72?

dagrayvid replied:
Good point

zvonkok commented on controllers/runtime.go:
```diff
@@ -144,6 +144,15 @@ func retryGetPushSecretName(r *SpecialResourceReconciler) (string, error) {
 
 func getPushSecretName(r *SpecialResourceReconciler) (string, error) {
```
in vanilla k8s there are no push secrets?

dagrayvid replied:
I don't think there are. At least not in my kOps cluster

dagrayvid committed:
Adding Platform field to RunInfo for cluster type, and using OnOCP where more specific checks don't make sense

dagrayvid committed:
Add make deploy-k8s and rearrange kustomize to use overlay for OCP/k8s

Commit:
Reorg of kustomize rules for kube and ocp

Added k8s e2e workflow

Simplified Makefile

Missing files

Removed not needed make target

Update workflow

Update manifests with Zvonko's changes

dagrayvid commented:
@zvonkok I rebased, and fixed the issue with the missing volume. Tested on vanilla and the e2e tests pass, so from my side this PR is ready to merge.

zvonkok commented:
/lgtm

zvonkok commented:
/lgtm

zvonkok commented:
/approve
